syntax = "proto3";

package mcubus.v1;

message RegisterRequest {
  string module_id = 1;
  string module_type = 2;
  map<string, string> metadata = 3;
}

message RegisterReply {
  bool success = 1;
  string assigned_id = 2;
  string message = 3;
}

message UnSubscribeRequest {
  string module_id = 1;
}

message UnSubscribeReplay {
  bool success = 1;
  string message = 2;
}

message SubscribeRequest {
  repeated string module_ids = 1;
  repeated string event_types = 2;
}
